frontend/lib/src/components/elements/PlotlyChart/PlotlyChart.tsx

```diff
@@ -167,23 +172,53 @@ function PlotlyFigure({
         // data is originalData + selectedpoints
         spec.data = data
         spec.layout.selections = selections
-
-        const hasSelectedPoints: boolean = spec.data.some(
-          (trace: any) =>
-            "selectedpoints" in trace && trace.selectedpoints.length > 0
-        )
-        if (hasSelectedPoints) {
-          spec.data.forEach((trace: any) => {
-            if (!trace.selectedpoints) {
-              // Plotly will automatically set traces in selectedpoints with the 100% opaqueness
-              // setting selectedpoints to an empty array will set these points to be opaque, thus representing that they're unselected
-              trace.selectedpoints = []
-            }
-          })
+        if (spec.data) {
+          const hasSelectedPoints: boolean = spec.data.some(
+            (trace: any) =>
+              "selectedpoints" in trace &&
+              trace.selectedpoints &&
+              trace.selectedpoints.length > 0
+          )
+          if (hasSelectedPoints) {
+            spec.data.forEach((trace: any) => {
+              if (!trace.selectedpoints) {
+                // Plotly will automatically set traces in selectedpoints with the 100% opaqueness
+                // setting selectedpoints to an empty array will set these points to be opaque, thus representing that they're unselected
+                trace.selectedpoints = []
+              }
+            })
+          }
         }
       }
     }
-
+    const rangeEvent = widgetMgr.getElementState(element.id, RANGE)
+    const autoRangeEvent = widgetMgr.getElementState(element.id, AUTORANGE)
+    const dragmode = widgetMgr.getElementState(element.id, DRAGMODE)
```

Comment:
Can you add a comment here to each of these variables describing what it contains

Reply:
Done